src/hoc/withTheme.js

```diff
@@ -23,25 +22,32 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     static styledComponentId = 'withTheme'
 
     static contextTypes = {
-      [CHANNEL]: PropTypes.func,
+      [CHANNEL]: CONTEXT_CHANNEL_SHAPE,
```

Comment:
@mxstbr, @philpl , moving convo re: breaking contextTypes change. 

TBF, I think we've all unintentionally relied on an implementation detail at some point or another ;). What about a `CHANNEL_NEXT` context for now and a deprecation warning issued when using the `unsubscribe` on `CHANNEL`?